sql/core/src/test/scala/org/apache/spark/sql/MiscFunctionsSuite.scala

```diff
@@ -56,6 +57,23 @@ class MiscFunctionsSuite extends QueryTest with SharedSparkSession {
       assert(e.getMessage.contains("current_user"))
     }
   }
+
+  test("SPARK-37591: AES functions - GCM mode") {
+    Seq(
+      ("abcdefghijklmnop", ""),
+      ("abcdefghijklmnop", "abcdefghijklmnop"),
+      ("abcdefghijklmnop12345678", "Spark"),
+      ("abcdefghijklmnop12345678ABCDEFGH", "GCM mode")
+    ).foreach { case (key, input) =>
+      val df = Seq((key, input)).toDF("key", "input")
+      val encrypted = df.selectExpr("aes_encrypt(input, key, 'GCM', 'NONE') AS enc", "input", "key")
+      assert(encrypted.schema("enc").dataType === BinaryType)
+      assert(encrypted.filter($"enc" === $"input").isEmpty)
+      val result = encrypted.selectExpr(
+        "CAST(aes_decrypt(enc, key, 'GCM', 'NONE') AS STRING) AS res", "input")
+      assert(!result.filter($"res" === $"input").isEmpty)
```

Comment:
To test all the records are decrypted correctly, should we do `assert(result.filter($"res" !== $"input").isEmpty)` ?

Reply:
Other AES tests are in `DataFrameFunctionsSuite`, so should we move this test to the same place?

Reply:
I would rather move other tests for misc functions from `DataFrameFunctionsSuite` to the dedicated `MiscFunctionsSuite`. I was thinking of moving all AES related tests here but decided to don't do unrelated changes.

Reply:
`DataFrameFunctionsSuite` has already had > 3800 lines. Don't think it makes sense to place new misc test there if there is a special test suite for such functions.